### PR TITLE
Reactivate binary stripping

### PR DESCRIPTION
**Pull request by isbm**
If there is no any particular reasons why we should not remove symbols information, then this should be activated back. However I do see the reason not to UPX-compress the release binaries by default, because it does not do any impact on memory-allocated size, however, it burns CPU to unpack the binary every launch, while disk is cheap. Some systems detect UPX compressed binaries as a virus, which only adds to the problem.

Whoever still needs UPX can use `make compress` as an extra step.

### CODE REVIEW

**Comment by Ichmed** [comment]
This is not due to the changes done here, but `cargo build` ignores profiles not defined in the top level cargo file

Since the release profiles are identical anyway the individual ones should be removed and replaced with one in the top Cargo.toml

**Comment by Ichmed** [comment]
I'm seconding removing the upx step since, afaik, it does not change binary size during runtime but only on disk.

Startup time is much more valuable then disk space (especially because many flakes may be starting at once while there will be only on pilot binary on disk)

File: firecracker-pilot/firecracker-service/service-communication/Cargo.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [profile.release]
```
Comment (Ichmed) [inline firecracker-pilot/firecracker-service/service-communication/Cargo.toml:1]: Move all profiles to the top level Cargo.toml

**Review by Ichmed — CHANGES_REQUESTED** [review]

File: firecracker-pilot/firecracker-service/service-communication/Cargo.toml (hunk 1 above)
Comment (schaefi) [reply firecracker-pilot/firecracker-service/service-communication/Cargo.toml:1]: yep would be good if along with this change all ```[profile.xxx]``` sections could be moved to the toplevel Cargo.toml. Messages like

```
warning: profiles for the non root package will be ignored, specify profiles at the workspace root:
```

will go away then

Thanks

**Comment by Ichmed** [comment]
LGTM

**Pull request merged**